src/Compress/GzAdapter.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Filter\Compress;
+
+use Laminas\Filter\Exception\ExtensionNotLoadedException;
+use Laminas\Filter\Exception\RuntimeException;
+
+use function assert;
+use function extension_loaded;
+use function gzcompress;
+use function gzdeflate;
+use function gzinflate;
+use function gzuncompress;
+
+/**
+ * Compression adapter for Gzip (ZLib)
+ *
+ * @psalm-type Options = array{
+ *     level?: int<0, 9>|null,
+ *     mode?: 'deflate'|'compress',
+ * }
+ */
+final class GzAdapter implements StringCompressionAdapterInterface
```

Comment:
As per the BZ2 adapter, the GZ adapter now only supports string content.

In future, it will be possible to introduce a `FileCompressionInterface` that this adapter could also implement so that we can filter an arbitrary string, or a single un-compressed file to `somefile.gz`